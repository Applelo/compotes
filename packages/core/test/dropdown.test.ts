import { afterAll, beforeAll, describe, expect, it } from 'vitest'
import type { Browser } from 'playwright'
import { chromium } from 'playwright'
import { destroyComponent } from './helper/destroy'

let browser: Browser

beforeAll(async () => {
  browser = await chromium.launch()
})

afterAll(async () => {
  browser.close()
})

describe('dropdown', async () => {
  it.concurrent('events', async () => {
    const page = await browser.newPage()
    const events = [
      'c.dropdown.opened',
      'c.dropdown.closed',
    ]
    const openedEvent = page.waitForEvent('console', msg => msg.text().includes('c.dropdown.opened'))
    const closedEvent = page.waitForEvent('console', msg => msg.text().includes('c.dropdown.closed'))
    await page.goto('http://localhost:3000/dropdown.html')

    await page.evaluate(() => {
      const el = document.querySelector('.c-dropdown')
      if (!el)
        return
      [
        'c.dropdown.opened',
        'c.dropdown.closed',
      ].forEach((e) => {
        el.addEventListener(e, () => {
          // eslint-disable-next-line no-console
          console.log(e)
        })
      })
    })

    const triggerBtn = page.locator('.c-dropdown-trigger').first()
    await triggerBtn.click()
    await triggerBtn.click()

    const result = (await Promise.all([
      openedEvent,
      closedEvent,
    ])).map(item => item.text())

    events.forEach(e => expect(result).toContain(e))
  })

  it.concurrent('destroy', async () => {
    const page = await browser.newPage()
    await page.goto('http://localhost:3000/dropdown.html')
    const { before, after } = await destroyComponent(page)
    expect(before).not.toEqual(after)
    expect(after).toMatchSnapshot()
  })
})
