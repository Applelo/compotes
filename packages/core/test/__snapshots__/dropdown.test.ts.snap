// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`dropdown > destroy 1`] = `
"<!DOCTYPE html><html lang="en"><head>
    <link rel="icon" href="data:image/svg+xml,<svg xmlns=%22http://www.w3.org/2000/svg%22 viewBox=%220 0 100 100%22><text y=%22.9em%22 font-size=%2290%22>üçØ</text></svg>">
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Dropdown</title>
    <script type="module" crossorigin="" src="/assets/dropdown-CePkjNFb.js"></script>
    <link rel="modulepreload" crossorigin="" href="/assets/_parent-PLeNLLls.js">
    <link rel="modulepreload" crossorigin="" href="/assets/index.esm-CjV0_wFH.js">
    <link rel="modulepreload" crossorigin="" href="/assets/accessibility-DQSUv1PD.js">
    <link rel="stylesheet" crossorigin="" href="/assets/marquee-ybAlwBfl.css">
  </head>
  <body>
    <div class="c-dropdown" style="margin-top: 2rem;">
      <button class="c-dropdown-trigger" aria-expanded="false" aria-haspopup="true">
        Basic Dropdown
      </button>
      <div class="c-dropdown-container" id="dropdown-container">
        Hello World
      </div>
    </div>
    <button class="js-destroy">
      Destroy
    </button>
  

</body></html>"
`;
